src/coreclr/tools/Common/InstructionSetHelpers.cs

```diff
@@ -93,16 +93,34 @@ internal static partial class Helpers
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("movbe");
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("popcnt");
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("lzcnt");
+                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("serialize");
 
                 // If AVX was enabled, we can opportunistically enable instruction sets which use the VEX encodings
                 Debug.Assert(InstructionSet.X64_AVX == InstructionSet.X86_AVX);
                 if (supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX))
                 {
+                    // TODO: Enable optimistic usage of AVX2 once we validate it doesn't break Vector<T> usage
+                    // optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("avx2");
```

Comment:
After having debugged some of the code in #85551, I'm fairly certain this is what was causing most of the issues.

In particular, `getJitFlags` includes any `OptimisticFlags` such as this and `ConvertToImpliedInstructionSetForVectorInstructionSets` would have converted `InstructionSet_VectorT256` to `InstructionSet_AVX2`, which would've then passed.

That does mean that `VectorT128/256/512` will need some different/special handling to get everything working as expected.